<?xml version="1.0" encoding="UTF-8"?>
<!-- Generated by org.testng.reporters.JUnitReportReporter -->
<testsuite hostname="Ruida-Xu" failures="1" tests="14" name="homework.ch11_13.CompositeComponentTest" time="0.000" errors="0" timestamp="07 4月 2022 05:55:23 GMT" skipped="0">
  <testcase classname="homework.ch11_13.CompositeComponentTest" name="testGetPrice" time="0.000"/>
  <testcase classname="homework.ch11_13.CompositeComponentTest" name="testRemove" time="0.000"/>
  <testcase classname="homework.ch11_13.CompositeComponentTest" name="testTestToString" time="0.000">
    <failure message="toString() 没有重用后台节点的toString方法 expected [true] but found [false]" type="java.lang.AssertionError">
      <![CDATA[java.lang.AssertionError: toString() 没有重用后台节点的toString方法 expected [true] but found [false]
at org.testng.Assert.fail(Assert.java:96)
at org.testng.Assert.failNotEquals(Assert.java:776)
at org.testng.Assert.assertTrue(Assert.java:44)
at homework.ch11_13.CompositeComponentTest.testTestToString(CompositeComponentTest.java:25)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:567)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
at org.testng.internal.Invoker.invokeMethod(Invoker.java:583)
at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:719)
at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:989)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
at org.testng.TestRunner.privateRun(TestRunner.java:648)
at org.testng.TestRunner.run(TestRunner.java:505)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:455)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:450)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:415)
at org.testng.SuiteRunner.run(SuiteRunner.java:364)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:84)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1208)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1204)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1137)
at org.testng.TestNG.runSuites(TestNG.java:1049)
at org.testng.TestNG.run(TestNG.java:1017)
at org.testng.TestNG.privateMain(TestNG.java:1354)
at org.testng.TestNG.main(TestNG.java:1323)
]]>
    </failure>
  </testcase> <!-- testTestToString -->
  <testcase classname="homework.ch11_13.CompositeComponentTest" name="testCalcPrice" time="0.000"/>
  <testcase classname="homework.ch11_13.CompositeComponentTest" name="testAdd" time="0.000"/>
  <testcase classname="homework.ch11_13.CompositeComponentTest" name="testIterator" time="0.000"/>
  <testcase classname="homework.ch11_13.CompositeComponentTest" name="testCalcPrice" time="0.000"/>
  <testcase classname="homework.ch11_13.CompositeComponentTest" name="testRemove" time="0.000"/>
  <testcase classname="homework.ch11_13.CompositeComponentTest" name="testGetPrice" time="0.000"/>
  <testcase classname="homework.ch11_13.CompositeComponentTest" name="testRemove" time="0.000"/>
  <testcase classname="homework.ch11_13.CompositeComponentTest" name="testAdd" time="0.000"/>
  <testcase classname="homework.ch11_13.CompositeComponentTest" name="testGetPrice" time="0.000"/>
  <testcase classname="homework.ch11_13.CompositeComponentTest" name="testAdd" time="0.000"/>
  <testcase classname="homework.ch11_13.CompositeComponentTest" name="testCalcPrice" time="0.000"/>
</testsuite> <!-- homework.ch11_13.CompositeComponentTest -->
